src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspQueryManager.java

```diff
@@ -146,6 +146,33 @@ private BuildTarget getBuildTargetForRule(Build.Rule rule) {
         .collect(Collectors.toList());
   }
 
+  public List<BuildTargetIdentifier> getTargetIdentifiersForDependencies(
+      List<BuildTargetIdentifier> targets) {
+    String targetsUnion =
```

Comment:
It looks like a good thing to move to the `TargetsUtils`

Reply:
the whole method or just getting the union? I'm not sure what you mean from your selection  :sweat_smile:

Reply:
this part: `targets.stream().map(BuildTargetIdentifier::getUri).collect(Collectors.joining("+"));`